### ISSUE TITLE
Android EAS build failing on Github Actions

### ISSUE THREAD

**Issue by alessandrocapra** [expo/eas-cli#1935]
### Build/Submit details page URL

https://expo.dev/accounts/quantoz/projects/qbs-test-app/builds/6708f5ba-dec0-4b4d-b474-440039bedffc#run-gradlew

## Notes
Building locally with the same setup works

## Expected
With the provided Github actions YML file below, a build should be created for both Android and iOS and the submission should take place thanks to the `--auto-submit` flag.

```

name: Build apps through Expo EAS 

on:
  pull_request:
    branches:
      - main

jobs:
  run-script:
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write

    steps:
      - name: Check for EXPO_TOKEN
        run: |
          if [ -z "${{ secrets.EXPO_TOKEN }}" ]; then
            echo "You must provide an EXPO_TOKEN secret linked to this project's Expo account in this repo's secrets. Learn more: https://docs.expo.dev/eas-update/github-actions"
            exit 1
          fi      
      - name: Check for GOOGLE_SERVICES_JSON
        run: |
          if [ -z "${GOOGLE_SERVICES_JSON}" ]; then
            echo "You must provide a GOOGLE_SERVICES_JSON environment variable for this job. This should be a secret linked to this project's Google service account in this repo's secrets."
            exit 1
          fi
        env:
          GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}
      
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Setup Google Service Account
        run: |
          echo '${{ secrets.GOOGLE_SERVICES_JSON }}' > ./mobile/google-service-account.json

      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18.x
          cache: npm
          cache-dependency-path: mobile/package-lock.json

      - name: Setup Expo and EAS
        uses: expo/expo-github-action@v8
        with:
          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Install dependencies
        run: |
          cd mobile
          npm install

      - name: Run script
        run: |
          cd mobile
          eas build --profile preview-store --platform all --non-interactive --auto-submit
```

## Actual behavior
iOS works perfectly, but the Android build fails with the error output below.

### Managed or bare?

Managed

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
      Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.0/bin/yarn
      npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
      Watchman: 2023.04.17.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.18 => 48.0.20
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.71.8 => 0.71.8
    npmGlobalPackages:
      eas-cli: 3.15.1
      expo-cli: 6.3.9
    Expo Workflow: managed

Expo doctor:
Check package.json for common issues
✔ Validate global prerequisites versions
✔ Validate Expo Config
✔ Check for conflicting global packages in project
✔ Check for incompatible packages
✔ Verify prebuild support package versions are compatible
✔ Check compatible dependency versions for the installed Expo SDK

Didn't find any issues with the project!

### Error output

```
Parsing json file: /home/expo/workingdir/build/mobile/android/app/google-services.json

> Task :expo-dev-launcher:generateReleaseAssets UP-TO-DATE

> Task :expo-constants:createReleaseExpoConfig

> Task :expo-dev-launcher:packageReleaseAssets

[stderr] 
FAILURE: Build completed with 2 failures.

[stderr] 
1: Task failed with an exception.

[stderr] 
-----------

[stderr] 
* What went wrong:

[stderr] 
Execution failed for task ':app:processReleaseGoogleServices'.

[stderr] 
> Missing project_info object

[stderr] 
* Try:

[stderr] 
> Run with --stacktrace option to get the stack trace.

[stderr] 
> Run with --info or --debug option to get more log output.

[stderr] 
> Run with --scan to get full insights.

[stderr] 
==============================================================================

[stderr] 
2: Task failed with an exception.

[stderr] 
-----------

[stderr] 
* What went wrong:

[stderr] 
java.lang.StackOverflowError (no error message)

[stderr] 
* Try:

[stderr] 
> Run with --stacktrace option to get the stack trace.

[stderr] 
> Run with --info or --debug option to get more log output.

[stderr] 
> Run with --scan to get full insights.

[stderr] 
==============================================================================

[stderr] 
* Get more help at https://help.gradle.org

[stderr] 
BUILD FAILED in 3m 49s
```

### Reproducible demo or steps to reproduce from a blank project

I think it might be something simple I am missing or that I could not find in the docs, if that's not the case I will try to create a new app and create the app on the google play console.

**Comment by szdziedzic** [comment]
Hi,

I googled it and it seems like [an issue connected with a missing `google-services.json` file](https://stackoverflow.com/questions/31825577/gradle-fails-building-with-missing-project-info-object).

Do you have push notifications set up for your project?

**Comment by alessandrocapra** [comment]
Thanks for the reply! Nope, no push notifications set up, I though EAS automatically takes the google-service-account.json file and transforms it into that one for the Android build process. 
I do not have such file that I manage myself, I am merely using the file I mentioned above to handle the submission through EAS Submit.

**Comment by alessandrocapra** [comment]
As additional info, I tried the following:
- setup a secret both on Github and Expo project, to store the contents of the JSON file (I have removed all unnecessary characters, like spaces and newlines)
- tried to encode the file contents in base64 and save that as secrets, then using the `eas-build-pre-install` hook to decode it back into the google-service-account.json file

**Comment by szdziedzic** [comment]
>Thanks for the reply! Nope, no push notifications set up, I though EAS automatically takes the google-service-account.json file and transforms it into that one for the Android build process.

So the EAS Build manages the Android build credentials needed for signing your build https://docs.expo.dev/app-signing/app-credentials/

The `google-services.json` file is something different. These credentials are needed for enabling Google APIs or Firebase services and are not managed by EAS. It seems like you aren't providing the `google-services.json` file as a `google-services.json` file, but some other service account credentials file.

The contents of `google-services.json` should look like
```json
{
  "project_info": {...},
  "client": [...],
}
```
https://developers.google.com/android/guides/google-services-plugin#processing_the_json_file

Are you sure that the file you provide has this structure?

**Comment by alessandrocapra** [comment]
Oh wait, I have a doubt now: it's the first time I use `--non-interactive`, does that respond yes to setting up push notifications? Because I usually always reply no, but manually.

**Comment by szdziedzic** [comment]
This question is asked only for iOS

**Comment by szdziedzic** [comment]
Are you using Firebase in your app?

**Comment by szdziedzic** [comment]
>Nope, not really. Maybe it's faster this way, is there a proven way to submit automatically when merging on main? I went through the docs and found some scattered instructions, including the Github Actions example, but it does not cover the example of submitting as well

So basically using the `--auto-submit` option (which you are already using) should do the job **if your build succeeds**. If your build fails the submission is canceled, because the `--auto-submit` option corresponds to the build you are starting to build using the same command.

**Comment by szdziedzic** [comment]
Did you add some new external dependencies between your last successful build and now?

**Comment by alessandrocapra** [comment]
Sorry, had other prios to work on. Back to this, I have only added some scripts for Github Actions, which should not be the cause for the failing build correct? 

Is there any command I could run to get more debug info on why is a google.services.json expected, since I am in the managed workflow and I do not even have android/ios folder in my project? 

Thanks for your help!

**Comment by alessandrocapra** [comment]
Silly me, I got confused by the guide in the environment variables section regarding EAS build, where they mentioned a google services file. I assumed it was the one I needed (google-service-account.json), but it's referring indeed to Firebase.

Sorry and thanks for helping out, I managed to save the contents of the Google Service Account file encoded as base64 in my Github repo secrets and in the CI step I am decoding it and writing it in the file. If there is a better way to do this, feel free to add a comment but it seems to work.
